client/components/domains/registrant-extra-info/uk-form.jsx

```diff
@@ -0,0 +1,192 @@
+		};
+		this.registrantTypeOptions = map( registrantTypes, ( text, optionValue ) => (
+			<option value={ optionValue } key={ optionValue }>
+				{ text }
+			</option>
+		) );
+	}
+
+	componentWillMount() {
+		// Add defaults to redux state to make accepting default values work.
+		const neededRequiredDetails = difference(
+			[ 'registrantType' ],
+			keys( this.props.contactDetailsExtra )
+		);
+
+		// Bail early as we already have the details from a previous purchase.
+		if ( isEmpty( neededRequiredDetails ) ) {
+			return;
+		}
+
+		this.props.updateContactDetailsCache( {
+			extra: pick( defaultValues, neededRequiredDetails ),
+		} );
+	}
+
+	handleChangeEvent = event => {
+		this.props.updateContactDetailsCache( {
+			extra: { [ camelCase( event.target.id ) ]: event.target.value },
+		} );
+	};
+
+	isTradingNameRequired( registrantType ) {
+		return includes(
+			[ 'LTD', 'PLC', 'LLP', 'IP', 'RCHAR', 'FCORP', 'OTHER', 'FOTHER', 'STRA' ],
+			registrantType
+		);
+	}
+
+	isRegistrationNumberRequired( registrantType ) {
+		return includes( [ 'LTD', 'PLC', 'LLP', 'IP', 'SCH', 'RCHAR' ], registrantType );
+	}
+
+	renderTradingNameField() {
+		const { contactDetails, translate } = this.props;
+		const { tradingName } = contactDetails.extra;
+		return (
+			<div>
+				<FormFieldset>
+					<FormLabel className="registrant-extra-info__trading-name" htmlFor="trading-name">
+						{ translate( 'Trading Name' ) }
+					</FormLabel>
+					<FormTextInput
+						id="trading-name"
+						value={ tradingName }
+						autoCapitalize="off"
+						autoComplete="off"
+						autoCorrect="off"
+						placeholder={ '' }
+						onChange={ this.handleChangeEvent }
+					/>
+				</FormFieldset>
+			</div>
+		);
+	}
+
+	renderRegistrationNumberField() {
+		const { contactDetails, translate } = this.props;
+		const { registrationNumber } = contactDetails.extra || '';
+		return (
+			<div>
+				<FormFieldset>
+					<FormLabel
+						className="registrant-extra-info__registration-number"
+						htmlFor="registration-number"
+					>
+						{ translate( 'Registration Number' ) }
+					</FormLabel>
+					<FormTextInput
+						id="registration-number"
+						value={ registrationNumber }
+						autoCapitalize="off"
+						autoComplete="off"
+						autoCorrect="off"
+						placeholder={ '' }
+						onChange={ this.handleChangeEvent }
+					/>
+				</FormFieldset>
+			</div>
+		);
+	}
+
+	render() {
+		const { translate } = this.props;
+		const { registrantType } = {
+			...defaultValues,
+			...this.props.contactDetails.extra,
+		};
+
+		return (
+			<form className="registrant-extra-info__form">
+				<p className="registrant-extra-info__form-desciption">
+					{ translate( 'Almost done! We need some extra details to register your %(tld)s domain.', {
+						args: { tld: '.uk' },
+					} ) }
+				</p>
+				<FormFieldset>
+					<FormLabel htmlFor="registrant-type">
+						{ translate(
+							'Choose the option that best describes your presence in the United Kingdom:'
```

Comment:
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Choose the option that best describes your Canadian presence:' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).